web/client/plugins/DeleteMap.jsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+import React from 'react';
+import PropTypes from "prop-types";
+import { Glyphicon } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import ConfirmDialog from '../components/misc/ConfirmDialog';
+import { createPlugin } from '../utils/PluginsUtils';
+import { deleteMap } from '../actions/maps';
+import { toggleControl } from '../actions/controls';
+import { mapIdSelector } from '../selectors/mapInitialConfig';
+import Message from '../components/I18N/Message';
+
+class DeleteConfirmDialog extends React.Component {
+
+    static propTypes = {
+        show: PropTypes.bool,
+        mapId: PropTypes.string,
+        onConfirmDelete: PropTypes.func,
+        onClose: PropTypes.func
+    };
+
+    static contextTypes = {
+        router: PropTypes.object
+    };
+
+    static defaultProps = {
+        show: false,
+        mapId: '',
+        onConfirmDelete: () => {},
+        onClose: () => {}
+    };
+
+    render() {
+        return (
+            <ConfirmDialog
+                show={this.props.show}
+                onClose={this.props.onClose}
+                title={<Message msgId="map.mapDelete" />}
+                onConfirm={() => {
+                    this.context.router.history.push("/");
+                    this.props.onConfirmDelete(this.props.mapId);
+                }}
+                fitContent
+            >
+                <div className="ms-detail-body">
+                    <Message msgId="resources.deleteConfirmMessage" />
+                </div>
+            </ConfirmDialog>
+        );
+    }
+}
+
+export const showConfirmDeleteMapModalSelector = state => state.controls && state.controls.mapDelete && state.controls.mapDelete.enabled;
```

Comment:
@ale-cristofori i meant to move this to the [controls selector](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/selectors/controls.js) file an import it here

and add a unit test [here](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/selectors/__tests__/controls-test.js)